lib/array.rb

```diff
@@ -0,0 +1,31 @@
+class Array
+  def gusject(*args, &block)
+    if args.empty? || args[0].instance_of?(Symbol)
+      gusject_no_initial(args, block)
+    else
+      gusject_with_initial(args, block)
+    end
+  end
+
+  private
+
+  def gusject_with_initial(args, block)
+    sum = args[0]
+    if block
+      each { |el| sum = block.call(sum, el) }
```

Comment:
Not sure if I understood the question correctly, but I will try to answer it.
Basically, \#gusject makes a decision of wether we've been given initial value or not. There is two possible combination for cases w/o initial value: block and no args OR symbol as single arg. Line 3 checks for both of these possibilities and if either is true we run \#gusject_no_initial. Instead of using drop(1).each I opted for setting sum to the first value of self. Then I simply iterate over the rest of self values using range like so: (1...length).each.
To be honest, I assumed that drop(1).each would mutate self and thus discarded that option. But after playing in IRB it now seems that it does not do that. So looks like a nice small refactor for the code ;)